pkg/scrape/scrape.go

```diff
@@ -464,6 +474,12 @@ mainLoop:
 					Value: l.Value,
 				})
 			}
+			for _, l := range sl.externalLabels {
```

Comment:
We need to maintain a sorted order, I think it would be better to add these labels before we call `sort.Sort` above.